keps/sig-security/2763-ambient-capabilities/README.md

```diff
@@ -0,0 +1,1066 @@
+3. Only the capability that is explicitly added using ambient field in the manifest will be added to the ambient set.
+4. Since the ambient capability set obeys the invariant that no capability can ever be ambient if it is not both permitted and inheritable, adding a capability to the ambient set will also add it to the permitted and inheritable set.
+
+#### Restricted ambient capabilities.
+
+Some capabilities like `CAP_SYS_ADMIN` and `CAP_DAC_OVERRIDE` make a non-root user very powerful and we should restrict them from being added to the ambient capabilities set.
+
+TODO(@vinayakankugoyal): List the capabilities that are restricted.
+
+### Changes to runtime API (https://pkg.go.dev/k8s.io/cri-api/pkg/apis/runtime/v1alpha2)
+
+We propose adding a new field to the [Capability](https://pkg.go.dev/k8s.io/cri-api/pkg/apis/runtime/v1alpha2#Capability) struct
+
+```diff
+type Capability struct {
+  // List of capabilities to add.
+  AddCapabilities []string `protobuf:"bytes,1,rep,name=add_capabilities,json=addCapabilities,proto3" json:"add_capabilities,omitempty"`
+  // List of capabilities to drop.
+  DropCapabilities     []string `protobuf:"bytes,2,rep,name=drop_capabilities,json=dropCapabilities,proto3" json:"drop_capabilities,omitempty"`
+  XXX_NoUnkeyedLiteral struct{} `json:"-"`
+  XXX_sizecache        int32    `json:"-"`
++ // List of ambient capabilities to add.
++ AddAmbientCapabilities []string `protobuf:"bytes,3,rep,name=add_ambient_capabilities,json=addAmbientCapabilities,proto3" json:"add_ambient_capabilities,omitempty"`
+}
+```
+
+Adding ambientCapabilities to the SecurityContext will directly update this field, and these changes would live in the [convertToRuntimeCapabilities](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/kuberuntime/security_context.go#L121:6) function.
+
+
+### Changes to containerd/containerd (https://github.com/containerd/containerd)
+
+`containerd` clears all ambient capabilities on container creation. See [this](https://github.com/containerd/containerd/blob/055c801ededcb7a5e82f47bdeed555cdf6c64bd8/pkg/cri/server/container_create_linux.go#L233) link for details. 
+
+```go
+// Clear all ambient capabilities. The implication of non-root + caps
+// is not clearly defined in Kubernetes.
+// See https://github.com/kubernetes/kubernetes/issues/56374
+// Keep docker's behavior for now.
+specOpts = append(specOpts,
+	customopts.WithoutAmbientCaps,
+	customopts.WithSelinuxLabels(processLabel, mountLabel),
+)
+```
+
+We would need to update code at that link to add the ambient capabilities by calling the [WithAmbientCapabilities](https://github.com/containerd/containerd/blob/ab963e1cc16a845567a0e3e971775c29c701fcf8/oci/spec_opts.go#L858) function instead of the [WithoutAmbientCapabilities](https://github.com/containerd/containerd/blob/04f73e3f8a097d95111f8419fa136d196b3a8725/pkg/cri/opts/spec_linux.go#L354) function. 
```

Comment:
Do we have an issue in containerd to track this proposal?

FYI @cpuguy83

Reply:
Yikes, I knew I was forgetting something! Thanks for calling this out. Ill go open one now.

Reply:
https://github.com/containerd/containerd/issues/5644

Reply:
I volunteer to do the work on containerd side. :)